sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceClientBuilder.java

```diff
@@ -146,6 +147,15 @@ public TableServiceAsyncClient buildAsyncClient() {
 
         AzureNamedKeyCredential namedKeyCredential = null;
 
+        // If 'endpoint' was provided, validate its format to end with the appropriate suffix
+        if (endpoint != null) {
+            String trimmedEndpoint = endpoint.endsWith("/") ? endpoint.substring(0,endpoint.length()-1) : endpoint;
```

Comment:
nit; CheckStyle might complain we're missing some spaces here.
```suggestion
            String trimmedEndpoint = endpoint.endsWith("/") ? endpoint.substring(0, endpoint.length() - 1) : endpoint;
```